docs/api/text/tsne.rst

```diff
@@ -20,27 +20,37 @@ After importing the required tools, we can :doc:`load the corpus <corpus>` and v
     corpus = load_corpus('hobbies')
     tfidf  = TfidfVectorizer()
 
-    docs   = tfidf.fit_transform(corpus.data)
-    labels = corpus.target
+    X   = tfidf.fit_transform(corpus.data)
+    y   = corpus.target
 
 Now that the corpus is vectorized we can visualize it, showing the distribution of classes.
 
 .. code:: python
 
     # Create the visualizer and draw the vectors
     tsne = TSNEVisualizer()
-    tsne.fit(docs, labels)
+    tsne.fit(X, y)
     tsne.poof()
 
 .. image:: images/tsne_all_docs.png
 
+Note that you can pass the class labels or document categories directly to the ``TSNEVisualizer`` as follows:
+
+.. code:: python
+
+    #Pass on a list of labels for the legend
+    labels = corpus.categories
+    tsne = TSNEVisualizer(labels=labels)
+    tsne.fit(X, y)
+    tsne.poof()
+
 If we omit the target during fit, we can visualize the whole dataset to see if any meaningful patterns are observed.
 
 .. code:: python
 
     # Don't color points with their classes
-    tsne = TSNEVisualizer(labels=["documents"])
-    tsne.fit(docs)
+    tsne = TSNEVisualizer(y=["documents"])
```

Comment:
Unfortunately, this is a small bug, it should be `labels=["documents"]`

Reply:
Yes,Thank you .I have made a PR.